### PR TITLE
container: T6210: add capability sys-nice

### PR DESCRIPTION
**Pull request by theflakes**
## Change Summary

Adding sys-nice as an option for container cap-add configuration. This is needed for Suricata v7.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [x] Other (please describe): Just adding a Vyos config option for functionality that already exists in podman.

## Related Task(s)

https://vyos.dev/T6210

## Component(s) name

container

## How to test

Configured container to run latest version of containerized Suricata. Running Vyos on my home router with Suricata v7 running as a container. Running on N305 Intel CPU with 32GB RAM.
```
container {
    name suricata {
        allow-host-networks
        arguments "-q 1 -q 2 -q 3 -q 4"
        cap-add net-admin
        cap-add sys-admin
        cap-add sys-nice
        image jasonish/suricata:latest
        memory 8192
        volume ETC {
            destination /etc/suricata
            source /config/suricata/etc
        }
        volume LOGS {
            destination /var/log/suricata
            source /config/suricata/logs
        }
        volume RULES {
            destination /var/lib/suricata
            source /config/suricata/rules
        }
    }
}
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [ ] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [ ] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by l0crian1** [comment]
Can you add it to the completionHelp list and add a valueHelp element?

**Comment by sever-sever** [comment]
Read please contributing document that you marked as “read”.
There are no tasks on phabricator, no task number in the commit message, no task number in the PR header.

We’ll close the  PR as it is not formatted correctly.

And addition you should use “commit —amend” and and “push —force” to override the previous commit

**Comment by c-po** [comment]
I wonder if we should take the last change in 1.4.0-epa3 to rename the node from `cap-add` to `capability` which is what it actually does, providing a system capability to the container.

**Comment by dmbaturin** [comment]
@c-po I tend to agree, let's rename it while we can. But I wonder if we should add a migration script to help people who used earlier versions of 1.4 pre-release builds.

@theflakes And, yes, please check out the contributing guide and follow the procedure. Not having corresponding tasks and not having commits and PRs linked to tasks makes release notes writing a real headache, it's important for tracking changes across multiple repositories.

**Comment by c-po** [comment]
> @c-po I tend to agree, let's rename it while we can. But I wonder if we should add a migration script to help people who used earlier versions of 1.4 pre-release builds.

Tracked via https://vyos.dev/T6208 - migrator will be implemented

**Comment by theflakes** [comment]
Sorry about that, I'm getting an account registered to then create the task.

File: interface-definitions/container.xml.in [hunk 1]
```diff
@@ -51,12 +51,16 @@
                 <format>sys-module</format>
                 <description>Load, unload and delete kernel modules</description>
               </valueHelp>
+              <valueHelp>
+                <format>sys-nice</format>
+                <description>Permissions to set process nice value</description>
```
Comment (c-po) [inline interface-definitions/container.xml.in]: Please drop the plural here

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/container.xml.in (hunk 1 above)
Comment (theflakes) [reply interface-definitions/container.xml.in]: done and thanks

**Comment by c-po** [comment]
@MergifyIo backport sagitta

**Pull request merged**